public/components/agents/syscollector/components/syscollector-table.tsx

```diff
@@ -117,19 +116,25 @@ export function SyscollectorTable({ tableParams }) {
           />
         </EuiFlexItem>
       </EuiFlexGroup>
-      {tableParams.exportFormatted &&
-        <EuiFlexGroup >
-          <EuiFlexItem grow={true} > </EuiFlexItem>
-          <EuiFlexItem grow={false} >
+      {tableParams.exportFormatted && tableParams.columns && (
+        <EuiFlexGroup>
+          <EuiFlexItem grow={true} />
+          <EuiFlexItem grow={false}>
             <EuiButtonEmpty
-              onClick={async () => await AppState.downloadCsv(tableParams.path, tableParams.exportFormatted, !!params.search ? [{ name: 'search', value: params.search }] : [])}
-              iconType="importAction">
+              onClick={async () =>
```

Comment:
better if this is in a separate function

Reply:
Do you say the onClick event?